contracts/test/FeeContract.t.sol

```diff
@@ -0,0 +1,336 @@
+import { Test } from "forge-std/Test.sol";
+import { OwnableUpgradeable } from
+    "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import { UUPSUpgradeable } from
+    "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+// Target contract
+import { FeeContract } from "../src/FeeContract.sol";
+import { DeployFeeContractScript } from "../script/DeployFeeContract.s.sol";
+
+/// @title FeeContract Test
+contract FeeContractTest is Test {
+    FeeContract public feeContract;
+
+    function setUp() public {
+        feeContract = new FeeContract();
+    }
+
+    //test deposits work
+    function testFuzz_deposit(address user, uint256 amount) public payable {
+        vm.assume(user != address(0));
+        amount = bound(amount, feeContract.MIN_DEPOSIT_AMOUNT(), feeContract.MAX_DEPOSIT_AMOUNT());
+
+        uint256 balanceBeforeUser = feeContract.getBalance(user);
+
+        //check that the depositEvent is emitted
+        vm.expectEmit(true, false, false, true);
+        // We emit the event we expect to see.
+        emit FeeContract.Deposit(user, amount);
+
+        //deposit for the two users
+        feeContract.deposit{ value: amount }(user);
+
+        //get the balance for that user after the deposit
+        uint256 balanceAfterUser = feeContract.getBalance(user);
+
+        //test that the users' balances have been incremented accurately
+        assertEq(balanceAfterUser, balanceBeforeUser + amount);
+
+        //test that the smart contract has the accumulative balance for both users
+        assertEq(address(feeContract).balance, amount);
+    }
+
+    // test depositing for many users
+    function test_depositForManyDifferentUsers() public payable {
+        for (uint256 i = 0; i < 10; i++) {
+            address user = makeAddr(string(abi.encode(i)));
+            uint256 amount = i / 100 ether + feeContract.MIN_DEPOSIT_AMOUNT();
+
+            //fund this account
+            vm.deal(user, amount);
+
+            //check the balance before
+            uint256 balanceBefore = feeContract.getBalance(user);
+
+            //prank as if the deposit is made by the user
+            vm.prank(user);
+
+            //deposit for the user
+            feeContract.deposit{ value: amount }(user);
+
+            //get the balance for that user after the deposit
+            uint256 balanceAfter = feeContract.getBalance(user);
+            assertEq(balanceAfter, balanceBefore + amount);
+        }
+    }
+
+    // test depositing for the same user many times
+    function test_depositManyTimesForTheSameUser() public payable {
+        address user = makeAddr("user");
+
+        //fund this account
+        vm.deal(user, 1 ether);
+
+        uint256 totalAmountDeposited = 0;
+
+        for (uint256 i = 0; i < 10; i++) {
+            uint256 amount = i / 100 ether + feeContract.MIN_DEPOSIT_AMOUNT();
+
```

Comment:
i think this will always be `MIN_DEPOSIT_AMOUNT` as the integer division will always round to zero?

Reply:
oh that's right!! the test still proves what it sets out to prove but I'll remove the division to make it easier https://github.com/EspressoSystems/espresso-sequencer/commit/b1884c4bea3246a86cbd430853a5a69e4def0f0a